app/src/components/blocks/_documentUpload/documentUploadLloydGeorgePreview/DocumentUploadLloydGeorgePreview.tsx

```diff
@@ -0,0 +1,89 @@
+import PDFMerger from 'pdf-merger-js/browser';
+import { UploadDocument } from '../../../../types/pages/UploadDocumentsPage/types';
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from 'react';
+import PdfViewer from '../../../generic/pdfViewer/PdfViewer';
+import Spinner from '../../../generic/spinner/Spinner';
+
+type Props = {
+    documents: UploadDocument[];
+    previewLoading: boolean;
+    setPreviewLoading: Dispatch<SetStateAction<boolean>>;
+    setMergedPdfBlob: Dispatch<SetStateAction<Blob | undefined>>;
+};
+
+const DocumentUploadLloydGeorgePreview = ({
+    documents,
+    previewLoading,
+    setPreviewLoading,
+    setMergedPdfBlob,
+}: Props) => {
+    const [mergedPdfUrl, setMergedPdfUrl] = useState('');
+
+    const runningRef = useRef(false);
+    useEffect(() => {
+        if (!documents || runningRef.current) {
+            return;
+        }
+
+        runningRef.current = true;
+
+        const render = async () => {
+            const merger = new PDFMerger();
+
+            for (const doc of documents) {
+                let attempts = 0;
+
+                do {
+                    try {
+                        await merger.add(doc.file);
+
+                        attempts = 3;
+                    } catch (err) {
+                        attempts += 1;
```

Comment:
have you noticed this to be require what sort of issues happen that require this?